### ISSUE TITLE
Formatters ran by apheleia doesn't respect per-project configuration

### ISSUE THREAD

**Issue by sirn** [raxod502/apheleia#1]
Thank you for making Apheleia!

I've found a little issue in the way Apheleia is using temporary file. Some formatter reads a configuration from a directory hierarchy until a file is found (e.g. `prettier` reads `.editorconfig`, and `black` reads `pyproject.toml`). Right now Apheleia uses `make-temp-file` which creates a temporary file in `/tmp`, which cause a per-project configuration to not get read during formatting.

### Steps to reproduce

1. Create an `.editorconfig` file with the following content:

    ```
    root = true
    
    [*]
    indent_size = 4
    ```

2. Create `test.js` with the following content:

    ```
    function foo() {
    return true;
    }
    ```

3. Format the file with `npx prettier test.js`
4. Format the file with Apheleia
5. Compare results between (3) and (4)

### Expected behavior

Apheleia-run formatted file becomes (4 spaces):

```
function foo() {
    return true;
}
```

### Actual behavior

Apheleia-run formatted file becomes (2 spaces):

```
function foo() {
  return true;
}
```

**Comment by raxod502** [comment]
Thank you for the kind and detailed issue report. I have pushed a commit which should fix the issue for Prettier. I am not sure if Black needs the actual filename, or if it is enough to call it with the project directory as the working directory. If the former, then this bug might be hard to fix, because I looked at the Black command-line options and there doesn't appear to be a way to tell Black to write the reformatted file to stdout. It only supports writing a diff to stdout, and I don't particularly want to try parsing that.

Let me know what you think.

**Comment by sirn** [comment]
Thank you for the fix! I can confirm that ad5f79f fixed the issue with `prettier`. I also tested with Black, which does seems to respect `default-directory`, e.g. with the following python file:

```
def foo():
    blah = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21]
```

and the following pyproject.toml:

```
[tool.black]
line-length = 80
```

…and the line properly get capped at 80 (instead of the default 88). So I think this issue is fixed. 🎉  Again, huge thanks for making Apheleia and fixing this issue!
